plugins/catalog-backend-module-gcp/src/providers/GkeEntityProvider.ts

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  TaskRunner,
+  readTaskScheduleDefinitionFromConfig,
+} from '@backstage/backend-tasks';
+import {
+  DeferredEntity,
+  EntityProvider,
+  EntityProviderConnection,
+} from '@backstage/plugin-catalog-node';
+
+import { Logger } from 'winston';
+import * as container from '@google-cloud/container';
+import {
+  ANNOTATION_KUBERNETES_API_SERVER,
+  ANNOTATION_KUBERNETES_API_SERVER_CA,
+  ANNOTATION_KUBERNETES_AUTH_PROVIDER,
+} from '@backstage/plugin-kubernetes-common';
+import { Config } from '@backstage/config';
+import { SchedulerService } from '@backstage/backend-plugin-api';
+
+/**
+ * Catalog provider to ingest GKE clusters
+ *
+ * @public
+ */
+export class GkeEntityProvider implements EntityProvider {
+  private readonly logger: Logger;
+  private readonly scheduleFn: () => Promise<void>;
+  private readonly gkeParents: string[];
+  private readonly clusterManagerClient: container.v1.ClusterManagerClient;
+  private connection?: EntityProviderConnection;
+
+  constructor(
+    logger: Logger,
+    taskRunner: TaskRunner,
+    gkeParents: string[],
+    clusterManagerClient: container.v1.ClusterManagerClient,
+  ) {
+    this.logger = logger;
+    this.scheduleFn = this.createScheduleFn(taskRunner);
+    this.gkeParents = gkeParents;
+    this.clusterManagerClient = clusterManagerClient;
+  }
+
+  public static fromConfig(
+    logger: Logger,
```

Comment:
maybe use an object here instead to make it easier to evolve.

Reply:
fixed